src/components/controls/filter.js

```diff
@@ -54,22 +55,38 @@ class FilterData extends React.Component {
      * each time a filter is toggled on / off.
      */
     const options = [];
-    Object.keys(this.props.activeFilters)
-      .forEach((filterName) => {
-        const filterTitle = this.getFilterTitle(filterName);
-        const filterValuesCurrentlyActive = this.props.activeFilters[filterName].filter((x) => x.active).map((x) => x.value);
-        Array.from(this.props.totalStateCounts[filterName].keys())
-          .filter((itemName) => isValueValid(itemName)) // remove invalid values present across the tree
-          .filter((itemName) => !filterValuesCurrentlyActive.includes(itemName)) // remove already enabled filters
-          .sort() // filters are sorted alphabetically - probably not necessary for a select component
-          .forEach((itemName) => {
-            options.push({
-              label: `${filterTitle} → ${itemName}`,
-              value: [filterName, itemName]
-            });
-          });
-      });
-    if (genotypeSymbol in this.props.activeFilters) {
+
+    /**
+     * First set of options is from the totalStateCounts -- i.e. every node attr
+     * which we know about (minus any currently selected filters). Note that we
+     * can't filter the filters to those set on visible nodes, as selecting a
+     * filter from outside this is perfectly fine in many situations.
+     *
+     * Those which are colorings appear first (and in the order defined in
+     * colorings). Within each trait, the values are alphabetical
+     */
+    const coloringKeys = Object.values(this.props.colorings||{}).map((c) => c.key);
+    const traitNames = Object.keys(this.props.totalStateCounts).sort((a,b) => {
+      return coloringKeys.indexOf(a) - coloringKeys.indexOf(b)
+    })
```

Comment:
If a trait is not included in `coloringKeys` then this will sort them to the beginning of the `traitNames` array. 

I think this needs to be updated to 
```suggestion
    const traitNames = Object.keys(this.props.totalStateCounts).sort((a,b) => {
      if(coloringKeys.indexOf(b) === -1) return -1;
      if(coloringKeys.indexOf(a) === -1) return 1;
      return coloringKeys.indexOf(a) - coloringKeys.indexOf(b)
    })
```

Reply:
Yeah, I've switched to a simpler to understand construction:

```js
    const coloringKeys = Object.keys(this.props.colorings||{});
    const unorderedTraitNames = Object.keys(this.props.totalStateCounts);
    const traitNames = [
      ...coloringKeys.filter((name) => unorderedTraitNames.includes(name)),
      ...unorderedTraitNames.filter((name) => !coloringKeys.includes(name))
    ]
```